src/corehost/common/utils.cpp

```diff
@@ -18,6 +18,12 @@ bool ends_with(const pal::string_t& value, const pal::string_t& suffix)
         (0 == value.compare(value.length() - suffix.length(), suffix.length(), suffix));
 }
 
+bool starts_with(const pal::string_t& value, const pal::string_t& prefix)
+{
+    return prefix.length() <= value.length() &&
+        (0 == value.compare(0, prefix.length(), prefix));
```

Comment:
You can use, `value.find(prefix, 0, prefix.length()) == 0` and the first check will be done by `find` to early out.